test/org/sagebionetworks/bridge/models/ClientInfoTest.java

```diff
@@ -15,229 +10,157 @@
+        assertClientInfo("appName/1.0 (deviceName; osName/4.0.0) sdkName/5.0.0", "appName", null, "deviceName",
+                "osName", "4.0.0", "sdkName", null);
         
-        info = ClientInfo.parseUserAgentString(VALID_LONG_UA_2_DEPRECATED);
-        assertEquals("Cardio Health", info.getAppName());
-        assertEquals(1, info.getAppVersion().intValue());
-        assertEquals(IOS, info.getOsName());
-        assertEquals("9.0.2", info.getOsVersion());
-        assertEquals("BridgeSDK", info.getSdkName());
-        assertEquals(4, info.getSdkVersion().intValue());
+        // Spacing does not break the parser
+        assertClientInfo("appName / 1 ( deviceName ; osName / 4) sdkName / 5", "appName", 1, "deviceName",
+                "osName", "4", "sdkName", 5);
         
-        info = ClientInfo.parseUserAgentString(VALID_LONG_UA_3_DEPRECATED);
-        assertEquals("Belgium", info.getAppName());
-        assertEquals(2, info.getAppVersion().intValue());
-        assertEquals(ANDROID, info.getOsName());
-        assertEquals("14", info.getOsVersion());
-        assertEquals("BridgeJavaSDK", info.getSdkName());
-        assertEquals(10, info.getSdkVersion().intValue());
-    }
-    
-    @Test
-    public void cacheWorks() {
-        ClientInfo info1 = ClientInfo.fromUserAgentCache(VALID_LONG_UA_1);
-        ClientInfo info2 = ClientInfo.fromUserAgentCache(VALID_LONG_UA_1);
+        // this can now get pretty wacky, we parse what we can, ignoring the rest
+        assertClientInfo("/3 (; iOS 10) SDK", null, 3, null, "iPhone OS", "10", "SDK", null);
+        assertClientInfo("/3 (; iOS 10)", null, 3, null, "iPhone OS", "10", null, null);
+        assertClientInfo("(;iOS 10) SDK", null, null, null, "iPhone OS", "10", "SDK", null);
+        assertClientInfo("/3 (;; iOS 10)", null, null, null, null, null, null, null);
+        assertClientInfo("/app (iOS) 10", null, null, null, "iPhone OS", null, null, 10);
+        assertClientInfo("/app", null, null, null, null, null, null, null);
+        assertClientInfo("10//10", null, null, null, null, null, null, null);
+        assertClientInfo("/7 (Android/10) /5", null, 7, null, "Android", "10", null, 5);
+        assertClientInfo("app/3 (Device; /2.0.0) sdk/4", "app", 3, "Device", null, "2.0.0", "sdk", 4);
+        assertClientInfo("AppName/1 (Device Name; iPhone OS) BridgeJavaSDK/3",
+                "AppName", 1, "Device Name", "iPhone OS", null, "BridgeJavaSDK", 3);
         
-        assertSame(info1, info2);
-    }
-    
-    @Test
-    public void cacheWorksWithBadValues() {
-        assertSame(ClientInfo.UNKNOWN_CLIENT, ClientInfo.fromUserAgentCache(null));
-        assertSame(ClientInfo.UNKNOWN_CLIENT, ClientInfo.fromUserAgentCache("   \n"));
-    }
-    
-    @Test
-    public void testIsSupportedAppVersion_GreaterThanSucceeds() {
-        ClientInfo info = ClientInfo.parseUserAgentString(VALID_LONG_UA_1);
-        assertTrue(info.isSupportedAppVersion(25));
```

Comment:
Do we need isSupportedAppVersion() tests?

Reply:
Yes I was overly zealous in deleting tests of methods besides the parsing method itself. Restored these.